desc/basis.py

```diff
@@ -1058,23 +1287,109 @@ def fourier(theta, m, NFP=1, dt=0):
         basis function(s) evaluated at specified points
 
     """
-    # for modest to large arrays, faster to find unique values and
-    # only evaluate those
-    unq_theta, tidx = np.unique(theta, return_inverse=True)
-    unq_m, midx = np.unique(m, return_inverse=True)
-
-    theta_2d = np.atleast_2d(unq_theta).T
-    m_2d = np.atleast_2d(unq_m)
-    m_pos = (m_2d >= 0).astype(int)
-    m_neg = (m_2d < 0).astype(int)
-    m_abs = np.abs(m_2d) * NFP
-    if dt == 0:
-        out = np.where(m_pos, np.cos(m_abs * theta_2d), np.sin(m_abs * theta_2d))
-    else:
-        out = m_abs * (m_neg - m_pos) * fourier(unq_theta, -unq_m, NFP=NFP, dt=dt - 1)
-    # put duplicate values back in
-    out = out[tidx][:, midx]
-    return out
+    return _fourier(theta, m, NFP, dt)
+
+
+@jit
+@jnp.vectorize
+def _fourier(theta, m, NFP, dt):
+    m_pos = (m >= 0).astype(int)
```

Comment:
On a platform without JAX installed, this line throws the error: `AttributeError: 'bool' object has no attribute 'astype'`.

Reply:
ahh yeah, its due to the different ways numpy vs jax vectorize a function. One other issue is that regular `np.vectorize` is really just a python for loop so would be pretty slow. I'll try to refactor things a bit to work better with regular numpy

Reply:
As for dedicated tests without JAX, I looked into it a while ago but its a bit tricky, since the decision on jax vs numpy is made when you import desc, so we'd have to run the test suite twice, with different environment variables